### PR TITLE
[BIOIN-417] Decreased memory requirements for NoGTReport and other pipelines

### PR DESCRIPTION
**Pull request by ilyasoifer**

### CODE REVIEW

**Review by doron-st — APPROVED** [review]
LGTM

File: ugvc/pipelines/train_models_pipeline.py [hunk 1]
```diff
@@ -159,13 +159,27 @@ def run(argv: list[str]):
...
+            "X_GCC",
+            "X_CSS",
+        ]
+
         with_dbsnp_bl = args.input_file.endswith("vcf.gz")
         if with_dbsnp_bl:
             if args.list_of_contigs_to_read != []:
-                dfs = [vcftools.get_vcf_df(args.input_file, chromosome=x) for x in args.list_of_contigs_to_read]
+                dfs = [
+                    vcftools.get_vcf_df(args.input_file, chromosome=x, ignore_fields=FIELDS_TO_IGNORE)
```
Comment (gilhornung) [inline ugvc/pipelines/train_models_pipeline.py:177]: Why is this related to NoGTReport?

File: ugvc/pipelines/run_no_gt_report.py [hunk 2]
```diff
@@ -265,8 +257,22 @@ def run_full_analysis(arg_values):
         [],
     )
 
+    FIELDS_TO_IGNORE = [
+        "GNOMAD_AF",
```
Comment (gilhornung) [inline ugvc/pipelines/run_no_gt_report.py:261]: I don't understand why GNOMAD_AF is ignored. We have an entire task dedicated to adding it! Maybe this task should be removed as well?

File: ugvc/vcfbed/vcftools.py [hunk 3]
```diff
@@ -27,11 +34,16 @@ def get_vcf_df(variant_calls: str, sample_id: int = 0, sample_name: str = None,
...
+    ignore_fields: list, optional
+        List of VCF tags to ignore (save memory)
     Returns
     -------
     pd.DataFrame
     """
+
+    if ignore_fields is None:
+        ignore_fields = []
+
```
Comment (gilhornung) [inline ugvc/vcfbed/vcftools.py:46]: You need to check if ignore_fields belong to the columns in the vcf, and inform the user if they are not

**Review by gilhornung — APPROVED** [review]

File: ugvc/pipelines/train_models_pipeline.py (hunk 1 above)
Comment (ilyasoifer) [reply ugvc/pipelines/train_models_pipeline.py:177]: And other pipelines :)

File: ugvc/pipelines/run_no_gt_report.py (hunk 2 above)
Comment (ilyasoifer) [reply ugvc/pipelines/run_no_gt_report.py:261]: We need this in MRD

File: ugvc/vcfbed/vcftools.py (hunk 3 above)
Comment (ilyasoifer) [reply ugvc/vcfbed/vcftools.py:46]: I don't think so, some are not required

**Pull request merged**